x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/index.tsx

```diff
@@ -323,9 +375,26 @@ export const CreatePackagePolicyPage: React.FunctionComponent = () => {
         history.push(packagePolicyPath);
       }
     },
-    [packagePolicy.policy_id, getPath, navigateToApp, history, routeState, wasNewAgentPolicyCreated]
+    [packagePolicy.policy_id, getPath, navigateToApp, history, routeState, queryParamsPolicyId]
   );
 
+  const createAgentPolicy = useCallback(async (): Promise<string | undefined> => {
```

Comment:
Looks like we do not have error handling here, and potentially we can promise unhandled

Reply:
there is error handling at the call site, this function is awaited and wrapped with try catch